packages/swing-store/src/snapStore.js

```diff
@@ -217,42 +196,125 @@ export function makeSnapStore(
 
   /** @param {string} hash */
   function fullPathFromHash(hash) {
-    return resolve(root, baseNameFromHash(hash));
+    return pathResolve(root, baseNameFromHash(hash));
   }
 
   /** @type { Set<string> } */
   const toDelete = new Set();
 
   /**
+   * Creates a new gzipped snapshot file in the `root` directory
+   * and reports information about the process,
+   * including file size and timing metrics.
+   * Note that timing metrics exclude file open/close/etc.
+   *
    * @param {(filePath: string) => Promise<void>} saveRaw
    * @returns {Promise<SnapshotInfo>}
    */
   async function save(saveRaw) {
-    return withTempName(async tmpSnapPath => {
-      const { duration: rawSaveSeconds } = await measureSeconds(() =>
+    const cleanup = [];
+    try {
+      // TODO: Refactor to use tmpFile rather than tmpName.
+      const tmpSnapPath = await ptmpName({ template: 'save-raw-XXXXXX.xss' });
+      cleanup.push(() => unlink(tmpSnapPath));
+      const { duration: rawSaveSeconds } = await measureSeconds(async () =>
         saveRaw(tmpSnapPath),
       );
       const { size: rawByteCount } = await stat(tmpSnapPath);
-      const { result: hash, duration: hashSeconds } = await measureSeconds(() =>
-        fileHash(tmpSnapPath),
-      );
+
+      // Perform operations that read snapshot data in parallel.
+      // We still serialize the stat and opening of tmpSnapPath
+      // and creation of tmpGzPath for readability, but we could
+      // parallelize those as well if the cost is significant.
+      const snapReader = createReadStream(tmpSnapPath);
+      cleanup.push(() => snapReader.destroy());
+      await fsStreamReady(snapReader);
+      // TODO: hoist.
```

Comment:
I'll be resolving these "hoist" TODOs while the PR goes through CI, along with updating now-redundant definitions for e.g. path vs. stream parameters. They were defined inline during development, and left in place for now to avoid delaying the PR (which can be merged with them depending upon urgency).